plugins/csv_dataset.cc

```diff
@@ -575,7 +575,13 @@ struct CsvDataset::Itl: public TabularDataStore {
         filename = config.dataFileUrl.toString();
         
         // Ask for a memory mappable stream if possible
-        ML::filter_istream stream(filename, { { "mapped", "true" } });
+        ML::filter_istream stream;
+        try {
+            stream.open(filename, { { "mapped", "true" } });
+        }
+        catch (const std::exception & e) {
+             throw HttpReturnException(400, "dataFileUrl pointed to an empty file");
```

Comment:
I would not make it fail with empty files.  In my experience, in real world situations they often occur, and really do have no rows.  Eg, if there was a data collection process that run hourly, but for some hours there was no activity.